tests/framework/microvm.py

```diff
@@ -917,6 +920,9 @@ def make_snapshot(
             net_ifaces=[x["iface"] for ifname, x in self.iface.items()],
             ssh_key=self.ssh_key,
             snapshot_type=snapshot_type,
+            meta={
+                "kernel_file": self.kernel_file,
```

Comment:
Now I don't quite remember, but there are some tests that read `kernel_file` and were not working when using the "restored" fixture. Now I don't know which test exactly was it. This was at the beginning but it has since changed substantially. I am not 100% sure it is needed anymore.

Reply:
I misinterpreted how the metadata works. I also used it in #4942 and now understand it! One drawback from the maintainability point of view is the difficulty to find whether the metadata is really needed or not, since, when it used, it is referenced through `MicroVM` instance. But it doesn't have to be addressed now and is not a blocker to merge this PR :)